cudax/include/cuda/experimental/__algorithm/common.cuh

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDAX_ALGORITHM_COMMON
+#define __CUDAX_ALGORITHM_COMMON
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__ranges/concepts.h>
+#include <cuda/std/__type_traits/is_convertible.h>
+#include <cuda/std/span>
+
+#include <cuda/experimental/__launch/launch_transform.cuh>
+
+namespace cuda::experimental
+{
+#if _CCCL_STD_VER >= 2020
+template <typename _Tp>
+concept __valid_copy_fill_argument = _CUDA_VRANGES::contiguous_range<detail::__as_copy_arg_t<_Tp>>;
+
+#else
+template <typename _Tp, bool = true>
+inline constexpr bool __convertible_to_span = false;
+
+template <typename _Tp>
+inline constexpr bool __convertible_to_span<
+  _Tp,
+  _CUDA_VSTD::is_convertible_v<_Tp, _CUDA_VSTD::span<typename _CUDA_VSTD::decay_t<_Tp>::value_type>>> = true;
```

Comment:
why not just
```suggestion
template <typename _Tp>
inline constexpr bool __convertible_to_span =
  _CUDA_VSTD::is_convertible_v<_Tp, _CUDA_VSTD::span<typename _CUDA_VSTD::decay_t<_Tp>::value_type>>;
```
?

Reply:
The main reason is that I don't want this to fail for a type without a `value_type` member.

For example in the testing for this PR I have a `weird_buffer` test type that is launch transformable to a span, but it has no `value_type`. With the current implementation it will evaluate `__valid_copy_fill_argument` to true and `copy_bytes` will accept it. If I simplify `__convertible_to_span` like that, it will fail to compile, because of that missing `value_type` member. There might be some simpler way to achieve that, but I couldn't find it.